### PR TITLE
Block Library: Mark new blocks to be included in WordPress core

### PR DESCRIPTION
**Pull request by gziolo**
## What?

Part of https://github.com/WordPress/gutenberg/issues/39889.

Blocks to be included in WordPress 6.0 release:

- Avatar
- Query No Results
- Read More
- Comment Author Name
- Comment Content
- Comment Date
- Comment Edit Link
- Comment Reply Link
- Comment Template
- Comments Query Loop
- Comments Pagination
- Comments Pagination Numbers
- Comments Pagination Previous

## Why?

We want to ship them to WordPress users in the 6.0 major release!

## How?

Block definitions were moved to `__experimentalGetCoreBlocks` function from `__experimentalRegisterExperimentalCoreBlocks` function.

### CODE REVIEW

**Comment by adamziel** [comment]
* https://github.com/WordPress/gutenberg/pull/40095 got closed
* This PR got opened with `discussionSettings`  renamed to `__experimentalDiscussionSettings`, but without the related JavaScript changes to make files like `packages/block-library/src/comment-template/hooks.js` consume that option

Would you please walk me through the reasoning here?

File: lib/compat/wordpress-6.0/blocks.php [hunk 1]
```diff
@@ -277,3 +277,56 @@ function gutenberg_block_type_metadata_multiple_view_scripts( $metadata ) {
...
+add_filter( 'block_editor_settings_all', 'gutenberg_extend_block_editor_settings_with_discussion_settings' );
+
+/**
+ * Mark the `children` attr of comments as embeddable so they can be included in
+ * REST API responses without additional requests.
+ *
+ * @return void
+ */
+function gutenberg_rest_comment_set_children_as_embeddable() {
+	add_filter(
```
Comment (gziolo) [inline lib/compat/wordpress-6.0/blocks.php:414]: @c4rl0sbr4v0 and @DAreRodz - is that still necessary for the Comment Template block?

**Comment by gziolo** [comment]
> * [Promote Avatar and Discussion settings to stable #40095](https://github.com/WordPress/gutenberg/pull/40095) got closed
> * This PR got opened with `discussionSettings`  renamed to `__experimentalDiscussionSettings`, but without the related JavaScript changes to make files like `packages/block-library/src/comment-template/hooks.js` consume that option
> 
> Would you please walk me through the reasoning here?

We don’t want to have the block editor setting fed with every possible global site setting. Discussion settings are used only when the comment-related blocks are inserted in the editor so we could use REST API instead. The challenge is that those settings are available through REST API for the admin users only. That's why we had to go with the fallback through the block editor settings. Besides, those aren't strictly editor settings but site settings.

File: lib/compat/wordpress-6.0/blocks.php (hunk 1 above)
Comment (cbravobernal) [reply lib/compat/wordpress-6.0/blocks.php:414]: We need that function in order to show Comment Replies on the editor. Attached a video of what happens if we remove it.

https://user-images.githubusercontent.com/37012961/162442445-462a64ae-3a62-471c-b117-68d122ce3864.mov

Comment (gziolo) [reply lib/compat/wordpress-6.0/blocks.php:414]: Great, thank you for the confirmation 👍🏻

**Comment by adamziel** [comment]
Ah alright! And the JS files from the other PR already rely on the __experimental name so there is nothing to change

**Review by adamziel — APPROVED** [review]

**Review by juanmaguitar — APPROVED** [review]

**Pull request merged**